tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -8978,6 +8978,48 @@ SpirvEmitter::processIntrinsicFirstbit(const CallExpr *callExpr,
                                        srcRange);
 }
 
+// Determines if the given expression is a valid output parameter to pass to
+// interlock* functions. Ideally, we could just check if the passed value is a
+// reference, but the HLSL frontend allows integers to be passed as output
+// parameters, without being marked as reference in the AST. This means we need
+// to walk the expression a bit to determine if it it a valid output parameter.
+// This function might miss some cases.
```

Comment:
Please clarify "miss some cases." Hopefully we could say something like

> Returns false if `expr` is definitely an invalid output parameter for the interlock functions. Note that `expr` might be invalid even if the function returns true.

Reply:
Reworded the whole comment to have a short usage sentence, and then the explanation around false-negatives.